### ISSUE TITLE
[Bug]: Getting Unexpected error: Invalid response 401 in Files app

### ISSUE THREAD

**Issue by ostasevych** [nextcloud/server#49187]
### ⚠️ This issue respects the following points: ⚠️

- [x] This is a **bug**, not a question or a configuration/webserver/proxy issue.
- [x] This issue is **not** already reported on [Github](https://github.com/nextcloud/server/issues?q=is%3Aopen+is%3Aissue+label%3Abug) OR [Nextcloud Community Forum](https://help.nextcloud.com/) _(I've searched it)_.
- [x] Nextcloud Server **is** up to date. See [Maintenance and Release Schedule](https://github.com/nextcloud/server/wiki/Maintenance-and-Release-Schedule) for supported versions.
- [x] I agree to follow Nextcloud's [Code of Conduct](https://nextcloud.com/contribute/code-of-conduct/).

### Bug description

After upgrade from 28 to 29 I've started to get Unexpected error: Invalid response 401 in Files app when trying to open any folder in Files webUI after leaving the webpage for a while. Clicking on Retry is not helpful. The only way is to manually reload the webpage.
![Image](https://github.com/user-attachments/assets/ae4adbaa-b5e3-4a13-a615-a3e953bdd1cc)

### Steps to reproduce

1. Open webUI files
2. Go somewhere
3. Leave it for a while: 3-6-12 hours
4. Click on any folder in the Files app
5. Observe Unexpected error: Invalid response 401 

### Expected behavior

The webpage should transparently renew without manual intrusion of a user, when a user comes back to it, similarly to Google Drive or any other cloud solution.

### Nextcloud Server version

29

### Operating system

Debian/Ubuntu

### PHP engine version

PHP 8.2

### Web server

Apache (supported)

### Database engine version

MariaDB

### Is this bug present after an update or on a fresh install?

Upgraded to a MAJOR version (ex. 28 to 29)

### Are you using the Nextcloud Server Encryption module?

None

### What user-backends are you using?

- [ ] Default user-backend _(database)_
- [ ] LDAP/ Active Directory
- [ ] SSO - SAML
- [ ] Other

### Configuration report

```json

```

### List of activated Apps

```shell

```

### Nextcloud Signing status

```shell

```

### Nextcloud Logs

```json

```

**Comment by joshtrichards** [comment]
I've never encountered this across numerous deployments. You'll need to do a bit more digging / give us more more details to sort of why you're encountering this.

What's the request that is failing in your browser inspector?
What about your config, such as session/logout parameters?
etc.

**Comment by ostasevych** [comment]
> I've never encountered this across numerous deployments. You'll need to do a bit more digging / give us more more details to help sort out why you're encountering this.
> 
> What's the request that is failing in your browser inspector? What about your config, such as session/logout parameters? Reverse proxy? etc.

I am facing with this issue every time after certain delay.
No reverse proxy, working behind VPN.

```
# cat config.php
<?php
$CONFIG = array (
  'instanceid' => '****',
  'passwordsalt' => '****',
  'secret' => '*****',
  'trusted_domains' =>
  array (
    0 => 'mycloud.tld',
    1 => 'office.mycloud.tld,
  ),
  'overwrite.cli.url' => 'https://mycloud.tld',
  'datadirectory' => '/nc/data,
  'dbtype' => 'mysql',
  'version' => '29.0.9.2',
  'dbname' => 'nextcloud',
  'dbhost' => 'localhost',
  'dbport' => '',
  'dbtableprefix' => 'oc_',
  'mysql.utf8mb4' => true,
  'dbuser' => 'server',
  'dbpassword' => '****',
  'installed' => true,
  'mail_smtpmode' => 'smtp',
  'mail_sendmailmode' => 'smtp',
  'mail_domain' => 'mail.com',
  'mail_send_plaintext_only' => false,
  'default_phone_region' => '**',
  'force_locale' => '**',
  'force_language' => '**',
  'allow_user_to_change_display_name' => false,
  'auth.webauthn.enabled' => false,
  'remember_login_cookie_lifetime' => 1296000,
  'session_lifetime' => 21600,
  'session_relaxed_expiry' => false,
  'session_keepalive' => false,
  'auto_logout' => false,
  'skeletondirectory' => '/nc/data/skeleton',
  'maintenance' => false,
  'maintenance_window_start' => 23,
  'sort_groups_by_name' => true,
  'memcache.local' => '\\OC\\Memcache\\APCu',
  'redis' =>
  array (
    'host' => '127.0.0.1',
    'port' => 6379,
  ),
  'memcache.locking' => '\\OC\\Memcache\\Redis',
  'app_install_overwrite' =>
  array (
    0 => 'files_fulltextsearch',
    1 => 'fulltextsearch_elasticsearch',
    2 => 'fulltextsearch',
    3 => 'files_fulltextsearch_tesseract',
    4 => 'twofactor_gateway',
    5 => 'files_rightclick',
    6 => 'twofactor_reminder',
  ),
  'activity_use_cached_mountpoints' => true,
  'mail_from_address' => 'cloud',
  'mail_smtpsecure' => 'ssl',
  'mail_smtpauthtype' => 'LOGIN',
  'mail_smtpauth' => 1,
  'mail_smtphost' => 'smtp.mail.com',
  'mail_smtpport' => '465',
  'mail_smtpname' => 'mycloud',
  'mail_smtppassword' => '*****',
  'theme' => '',
  'logtype' => 'file',
  'logfile' => '/nc/data/nextcloud.log',
  'loglevel' => 2,
  'loglevel_frontend' => 2,
  'log_rotate_size' => 104857600,
  'trashbin_retention_obligation' => 'auto, 30',
  'logtimezone' => 'Europe/***',
  'filesystem_check_changes' => 0,
  'profile.enabled' => false,
  'enable_previews' => true,
  'preview_format' => 'webp',
  'preview_max_x' => '2048',
  'preview_max_y' => '2048',
  'enabledPreviewProviders' =>
  array (
    0 => 'OC\\Preview\\Image',
    1 => 'OC\\Preview\\PNG',
    2 => 'OC\\Preview\\JPEG',
    3 => 'OC\\Preview\\WebP',
  ),
  'jpeg_quality' => '75',
  'preview_concurrency_new' => '1',
  'twofactor_enforced' => 'true',
  'allow_local_remote_servers' => true,
  'has_internet_connection' => true,
  'defaultapp' => '',
);
```

**Comment by susnux** [comment]
This sounds like the csrf token is not updated.
Have you tried to clear all browser cache after the update?

Could you send the browser dev tools outputs?

**Comment by susnux** [comment]
> Likely why. In which case this seems expected behavior based on your configuration.

Makes sense!

**Comment by Hawk7ns** [comment]
Hello, 
I have the same issue with one of the nextcloud account, I have deleted totally this account and when recreating it with the same login the issue came back. So I created a new one with different name and it solve the issue. BUT it came back this morning after connecting this account on 3 devices at the same time.

**Comment by pltcldvlpr** [comment]
Can confirm, same notification in the files app. Syncing is not affected, the files are still on server and local machines. After multiple reloads or clicking around, files appear as expected

Ubuntu 24.04.1 LTS, Nginx, MariaDB, Nextcloud 30.0.2

**Comment by LucasISC** [comment]
the issue happens to me only on firefox, when I try on chrome everything works fine. Also this issue is relatedto broken sharing button and the impossibility to logout (error message : "csrf check failed")

Debian, Nginx, MariaDB, Nextcloud 30.0.2, NC Docker

**Comment by toxpal** [comment]
Same issue here for months when using Files app.

**Comment by susnux** [comment]
Should be fixed with Nextcloud 29.0.12 and 30.0.6.
If you encounter a similar issue **after** updating to at least one of those versions, please open a new issue.
